html_css/v2/main-course/transforms-animation.md

```diff
@@ -0,0 +1,176 @@
+
+CSS transitions were designed to animate an element from one state to another. They *can* loop but they weren't designed for that, whereas keyframes were designed with the purpose of explicitly enabling loops.
+
+Transitions need a trigger (remember the pseudo-classes?) but keyframes do not. Once you have your elements in place and CSS defined, an animation will start running immediately  if that's what you told it to do. 
+
+Transitions are also not as flexible as using keyframes. When you define a transition, imagine you are sending that element on a journey in a straight line from point A to point B. Yes the `transition-timing-function` can add some variance to the timing of this change, but it doesn't compare to the amount of flexibility added by using keyframes.
+
+There are more differences between the methods than those described above, but hopefully you have a clearer idea of the differences between them now. If you need to change the opacity of an element when it is active, then a keyframe animation would be overkill. But if you need to carry out something more complicated, keyframes provide you with the tools you need. 
+
+Let's look at some keyframe animation syntax. Like transitions, there is a long and short form of using animations. We have used the long form here so you can get to grips with some of the main animation sub-properties that you will use to get started. 
+
+~~~css
+#ball {
+  /* ... other css properties ... */
+  animation-duration: 3s;
+  animation-name: bounce;
+  animation-iteration-count: infinite;
+  animation-direction: alternate;
+}
+
+@keyframes bounce {
+  from {
+    margin-top: 100%;
+  }
+
+  to {
+    margin-top: 0%;
+  }
+}
+~~~
+
+Note that this process has two distinct parts. The first part is the configuration stage where we define our animation sub-properties on the `#ball` element. Here we have an `animation-duration` of three seconds. This means that it will take three seconds for the `#ball` element to complete one animation cycle. Next we have defined the `animation-name` to be "bounce" which is essential for the `@keyframe` section coming up next. This is just a custom name that is not a particular CSS value. We could have called it "pineapples" if we so wished, but for our purposes "bounce" suits well. Then our `animation-iteration-count` is set to `infinite` which means this animation will run forever. You could set this to 1, 2, or as many iterations as you wish. Finally our `animation-direction` decides if our animation should alternate direction on the completion of one cycle, or reset to the start point and repeat itself. Here we set it to `alternate`, which means that the `#ball` will smoothly animate back along the path it came from in reverse as opposed to skipping straight back to its original point.  
```

Comment:
as above... I think this info is fine, but that it would be better consumed as a bulleted list.